cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedules.java

```diff
@@ -64,4 +64,87 @@ private Schedules() {
   public enum Source {
     STREAM
   }
+
+  /**
+   * Get a builder used to create a schedule.
+   *
+   * @param name the name of the schedule to build
+   * @return a builder used to create a schedule
+   */
+  public static Builder builder(String name) {
+    return new Builder(name);
+  }
+
+  /**
+   * Builder used to create a schedule.
+   */
+  public static class Builder {
+    private final String name;
+    private String description;
+    private Integer concurrentProgramRunsThreshold;
+
+    private Builder(String name) {
+      this.name = name;
+      this.description = "";
+    }
+
+    /**
+     * Set the schedule description.
+     *
+     * @param description the schedule description
+     * @return the Builder
+     */
+    public Builder setDescription(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Skip running the scheduled program if the number of concurrent runs of the program exceeds the specified
+     * threshold. Prior to running the program, the scheduler will check how many runs of the program are currently
+     * in the RUNNING state.
+     * If that number is greater than the specified threshold, the scheduler will not run the program.
+     * This number includes runs of the program that were started outside of this schedule,
+     * either manually or by another schedule. For example, if the threshold is set to 0, the scheduler will
+     * only run the program if there are no other runs of the program currently in the RUNNING state.
+     *
+     * If no threshold is set, there will be no limit on the number of concurrent runs.
+     *
+     * @param threshold skip the scheduled run if the number of concurrent program runs is above this threshold
+     * @return the Builder
+     */
+    public Builder skipIfConcurrentProgramRunsExceed(int threshold) {
```

Comment:
Is there also a way to defer the new run until the concurrency is below the threshold? Or does it have to be skipped?